ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OGridSearch.scala

```diff
@@ -390,14 +392,14 @@ trait H2OGridSearchParams extends H2OCommonParams with Params {
   private val algo = new DoubleParam(this, "algo", "dummy argument for pysparkling")
   protected final val gridAlgoParams = new AlgoParams(this, "algoParams", "Specifies the algorithm for grid search")
   private val hyperParameters = new HyperParamsParam(this, "hyperParameters", "Hyper Parameters")
-  private val strategy = new GridSearchStrategyParam(this, "strategy", "Search criteria strategy")
+  private val strategy = new Param[String](this, "strategy", "Search criteria strategy")
   private val maxRuntimeSecs = new DoubleParam(this, "maxRuntimeSecs", "maxRuntimeSecs")
   private val maxModels = new IntParam(this, "maxModels", "maxModels")
   private val stoppingRounds = new IntParam(this, "stoppingRounds", "Early stopping based on convergence of stoppingMetric")
   private val stoppingTolerance = new DoubleParam(this, "stoppingTolerance", "Relative tolerance for metric-based" +
     " stopping criterion: stop if relative improvement is not at least this much.")
-  private val stoppingMetric = new StoppingMetricParam(this, "stoppingMetric", "Stopping Metric")
-  private val selectBestModelBy = new MetricParam(this, "selectBestModelBy", "Select best model by specific metric." +
+  private val stoppingMetric = new Param[String](this, "stoppingMetric", "Stopping Metric")
+  private val selectBestModelBy = new NullableStringParam(this, "selectBestModelBy", "Select best model by specific metric." +
```

Comment:
Wouldn't be `Param[String]` sufficient if the default value is `"AUTO"`?

Reply:
👍 great catch!